pkg/operator/controller/dns_status.go

```diff
@@ -0,0 +1,146 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// syncDNSStatus computes the current status of dns and
+// updates status upon any changes since last sync.
+func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain string, ds *appsv1.DaemonSet) error {
+	updated := dns.DeepCopy()
+	updated.Status.ClusterIP = clusterIP
+	updated.Status.ClusterDomain = clusterDomain
+	updated.Status.Conditions = computeDNSStatusConditions(updated.Status.Conditions, clusterIP, ds)
```

Comment:
We do want to pass the old status conditions, `computeDNSStatusConditions(dns.Status.Conditions,..`  might be clear?

Reply:
`setDNSStatusCondition` creates a slice of conditions. The func has logic to append conditions to the new slice and return it. I thought we should be operating on a copy (i.e. `updated`) of the object under these circumstances?